In [1]:
from ultralytics import YOLO
import pandas as pd
import onnxruntime as ort
import cv2
import numpy as np
from matplotlib import pyplot as plt
from sklearn import metrics

from pathbook.pathbook import *

In [2]:
results = YOLO('../bin/yolov8x-seg.onnx').predict(path_test_dataset,imgsz=320,batch=10)
df = pd.read_csv(path_test_annotation)

Loading ../bin/yolov8x-seg.onnx for ONNX Runtime inference...

image 1/595 /Users/samedi/Documents/Coding/swans/dataset/test/0.jpg: 320x320 1 shipun, 321.6ms
image 2/595 /Users/samedi/Documents/Coding/swans/dataset/test/1.jpg: 320x320 11 maliys, 284.4ms
image 3/595 /Users/samedi/Documents/Coding/swans/dataset/test/10.jpg: 320x320 3 maliys, 286.2ms
image 4/595 /Users/samedi/Documents/Coding/swans/dataset/test/100.jpg: 320x320 1 klikun, 276.6ms
image 5/595 /Users/samedi/Documents/Coding/swans/dataset/test/101.jpg: 320x320 5 klikuns, 301.9ms
image 6/595 /Users/samedi/Documents/Coding/swans/dataset/test/102.jpg: 320x320 1 shipun, 269.4ms
image 7/595 /Users/samedi/Documents/Coding/swans/dataset/test/103.jpg: 320x320 1 maliy, 270.0ms
image 8/595 /Users/samedi/Documents/Coding/swans/dataset/test/104.jpg: 320x320 1 shipun, 267.4ms
image 9/595 /Users/samedi/Documents/Coding/swans/dataset/test/105.jpg: 320x320 1 shipun, 304.0ms
image 10/595 /Users/samedi/Documents/Coding/swans/dataset/test/106.j

In [3]:
y_pred = []
y_true = []

for result in results:
    bboxes = pd.DataFrame(result.boxes.data, columns=['x1','y1','x2','y2','conf','class'])
    y_pred.append(result.names[bboxes.iloc[bboxes['conf'].idxmax()]['class']])

    answer_df = df[df.path == result.path.split('/')[-1]]['label']
    y_true.append(answer_df.iloc[answer_df.count().argmax()])

print(metrics.accuracy_score(y_true, y_pred))

0.9193277310924369 0.9193277310924369


              size   time   accuracy    
yolov7         640    270     0.9126        
yolov8n-seg    320     20     0.8907
yolov8s-seg    640    174     0.9294
yolov8x-seg    320    386     0.9193